testing/sdk_plan.py

```diff
@@ -136,10 +143,20 @@ def wait_for_plan_status(
     else:
         statuses = status
 
+    initial_failures = sum(sdk_metrics.get_failure_metrics(service_name).values())
+
     @retrying.retry(
-        wait_fixed=1000, stop_max_delay=timeout_seconds * 1000, retry_on_result=lambda res: not res
+        wait_fixed=1000,
+        stop_max_delay=timeout_seconds * 1000,
+        retry_on_result=lambda res: not res,
+        retry_on_exception=lambda e: not isinstance(e, FailuresExceededError),
     )
     def fn():
+        failure_metrics = sdk_metrics.get_failure_metrics(service_name)
+        failures = sum(failure_metrics.values())
+        if failures - initial_failures > ALLOWED_FAILURES:
+            raise FailuresExceededError("Service not recoverable: {}".format(service_name))
```

Comment:
I would add more context to the exception message, or at the very least add an `.error` log line.